sdk/tests/router/router_test.py

```diff
@@ -268,6 +268,56 @@ def test_list_versions(turing_api, active_project, generic_router, generic_route
         assert actual.updated_at == generic_router_version.updated_at
 
 
+@responses.activate
+def test_create_version(turing_api, active_project, generic_router_config, generic_router_version, use_google_oauth):
+    turing.set_url(turing_api, use_google_oauth)
+    turing.set_project(active_project.name)
+
+    base_router = turing.Router(
+        id=191,
+        name="router-1",
+        project_id=active_project.id,
+        environment_name="id-dev",
+        monitoring_url="http://localhost:5000/endpoint_1",
+        status=turing.router.config.router_version.RouterStatus.DEPLOYED,
+    )
+
+    responses.add(
+        method="POST",
+        url=f"/v1/projects/{active_project.id}/routers/{base_router.id}/versions",
+        body=json.dumps(generic_router_version, default=tests.json_serializer),
+        status=200,
+        content_type="application/json"
+    )
+
+    actual_response = base_router.create_version(generic_router_config)
```

Comment:
Why are we creating a router here for the comparison? Why can we not simply compare using a `RouterVersion` entity?

Reply:
I think this is linked to your observation in the comment below that `create_version` is a method of the `Router` class; creating a version requires a base router object to exist so that's why the test required creating a router. But of course, if we were to move `create_version` to a method of `RouterVersion` then there'll be no need to create a router here.